sources/packages/backend/apps/api/src/services/student-file/student-file.service.ts

```diff
@@ -44,24 +48,35 @@ export class StudentFileService extends RecordDataModelService<StudentFile> {
     studentId: number,
     auditUserId: number,
   ): Promise<StudentFile> {
+    try {
```

Comment:
Can we refactor the method as below to use the `ProcessSummary` only?
Right now it is using the logger and `ProcessSummary`.
```ts
async createFile(
    createFile: CreateFile,
    studentId: number,
    auditUserId: number,
  ): Promise<StudentFile> {
    const summary = new ProcessSummary();
    try {
      summary.info(`Uploading file ${createFile.fileName} to S3 storage.`);
      // File upload to the S3 file storage.
      await this.uploadFileToStorage(createFile);
      summary.info(`File ${createFile.fileName} uploaded to S3 storage.`);
    } catch (error: unknown) {
      summary.error(`Error while uploading ${createFile.fileName}.`, error);
      this.logger.logProcessSummary(summary);
      throw new CustomNamedError(
        `Unexpected error while uploading the file ${createFile.fileName}.`,
        FILE_SAVE_ERROR,
      );
    }
    const newFile = new StudentFile();
    newFile.fileName = createFile.fileName;
    newFile.uniqueFileName = createFile.uniqueFileName;
    newFile.groupName = createFile.groupName;
    newFile.student = { id: studentId } as Student;
    newFile.creator = { id: auditUserId } as User;
    newFile.virusScanStatus = VirusScanStatus.InProgress;

    summary.info(`Saving the file ${createFile.fileName} to database.`);
    let savedFile: StudentFile;
    try {
      savedFile = await this.repo.save(newFile);
    } catch (error: unknown) {
      summary.error(`Error saving the file: ${error}`, error);
      this.logger.logProcessSummary(summary);
      throw new CustomNamedError(
        `Unexpected error while uploading the file ${newFile.fileName}.`,
        FILE_SAVE_ERROR,
      );
    }
    // Add to the virus scan queue only if the file is successfully saved to the database.
    try {
      summary.info(
        `Adding the file: ${newFile.fileName} to the virus scan queue.`,
      );
      await this.virusScanQueue.add({
        uniqueFileName: createFile.uniqueFileName,
        fileName: createFile.fileName,
      });
      summary.info(
        `File ${newFile.fileName} has been added to the virus scan queue.`,
      );
      return savedFile;
    } catch (error: unknown) {
      // If adding the file to the virus scanning queue fails,
      // then revert the file virus scan status in the database to pending.
      summary.error(
        `Error while enqueueing the file ${newFile.fileName} for virus scanning.`,
        error,
      );
      await this.repo.update(
        { uniqueFileName: newFile.uniqueFileName },
        { virusScanStatus: VirusScanStatus.Pending },
      );
    } finally {
      this.logger.logProcessSummary(summary);
    }
  }
```